llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -5067,6 +5067,62 @@ Instruction* cs6475_optimizer(Instruction *I) {
   }
   // END JOHN REGEHR
 
+  //BEGIN ZEYUAN WANG
+  // x*(x+2) + 1 -> (x+1)*(x+1)
+  Value *X1 = nullptr;
```

Comment:
please put your optimization inside of a { } block so that your local variable names do not escape. everyone is using the same variable names.